profile/middle_weight_benchmark.rb

```diff
@@ -0,0 +1,233 @@
+
+  tms_results = []
+  reps.times do
+    database.drop
+    tms_results = Benchmark.bm do |bm|
+      bm.report('Middleweight::Large doc bulk insert') do
+        collection.insert_many(large_doc_data, ordered: true)
+      end
+    end
+  end
+
+  # Get the real times (wall clock times) from the Benchmark::Tms objects
+  return tms_results.map { |result| result.real }, data_file_size/1000000.0, "Large doc bulk insert"
+end
+
+
+##
+# GridFS upload
+#
+# - Drop the database.
+# - Load the GRIDFS_LARGE file as a string.
+#
+# Measure: Upload the GRIDFS_LARGE data as a GridFS file 100 times.  Each time, use a
+#          different filename, following the sprintf pattern "gridfstest%03d" with integers
+#          from 1 to 100.
+#
+# @param [ Mongo::Database ] database MongoDB Database.
+# @param [ Integer ] reps Number of repetitions of the benchmark to run.
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def gridfs_upload(database, reps)
+  gridfs_large_data_string = BenchmarkHelper.load_string_from_file(GRIDFS_LARGE)
+  data_file_size = File.size(GRIDFS_LARGE)
+
+  tms_results = []
+  reps.times do
+    database.drop
+    tms_results << Benchmark.bm do |bm|
+      bm.report('Middleweight::GridFS upload') do
+        100.times do |i|
+          database.fs.insert_one(
+              Mongo::Grid::File.new(gridfs_large_data_string, :filename => "gridfstest%03d" % (i+1))
+          )
+        end
+      end
+    end.first
+  end
+
+  # Get the real times (wall clock times) from the Benchmark::Tms objects
+  return tms_results.map { |result| result.real }, data_file_size*100/1000000.0, "GridFS upload"
+end
+
+
+# GridFS download
+#
+# - Drop the database
+# - Upload the GRIDFS_LARGE file to the default gridFS bucket with the name "gridfstest".
+#   Record the _id of the uploaded file.
+#
+# Measure: Download the "gridfstest" file by its _id 100 times.  Use whatever download API
+#         is most natural for each language (e.g. open_download_stream(), read from the
+#         stream into a variable).  Discard the downloaded data after each download.
+#
+# @param [ Mongo::Database ] database MongoDB Database.
+# @param [ Integer ] reps Number of repetitions of the benchmark to run.
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def gridfs_download(database, reps)
+  database.drop
+  gridfs_large_data_string = BenchmarkHelper.load_string_from_file(GRIDFS_LARGE)
+  data_file_size = File.size(GRIDFS_LARGE)
+  gridfs_file = Mongo::Grid::File.new( gridfs_large_data_string, :filename => GRIDFSTEST )
+  database.fs.insert_one( gridfs_file )
+
+  tms_results = Benchmark.bm do |bm|
+    reps.times do
+      bm.report('Middleweight::GridFS download') do
+        100.times do
+          stream = StringIO.new
```

Comment:
maybe figure out a way to not have the instantiation of a new stream factor into the timing